requestParser/requestParser.go

```diff
@@ -181,6 +182,19 @@ func (r *RequestParser) parseRequests() {
 func (r *RequestParser) parseRequest(request WrpWithTime) {
 	defer r.parseWorkers.Release()
 
+	//use regex matching to see what event type event is, for events metrics
+	eventDestination := getEventDestinationType(request.Message.Destination)
+
+	//If partner ID exists, grab first partner ID for event metrics
+	var partnerID string
+	if request.Message.PartnerIDs != nil && len(request.Message.PartnerIDs) > 0 {
+		partnerID = request.Message.PartnerIDs[0]
+	} else {
+		partnerID = noPartnerID
+	}
```

Comment:
Ideally, we want the partner ID value to be determined the same way as this function:
https://github.com/xmidt-org/webpa-common/blob/main/basculechecks/metricvalidator.go#L123

I think the best bet would be to make this function public, rather than duplicating it.  I'll work on that, so you can call it here.